src/main/java/com/example/booking_app/service/SeatsServiceImplementation/CinemaSeatServiceImplementation.java

```diff
@@ -0,0 +1,233 @@
+package com.example.booking_app.service.SeatsServiceImplementation;
+
+import com.example.booking_app.model.*;
+import com.example.booking_app.repository.CinemaSeatRepository;
+import com.example.booking_app.service.CinemaSeatService;
+import com.example.booking_app.service.ReservationService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import java.time.LocalDateTime;
+import java.time.LocalTime;
+import java.util.*;
+import java.util.stream.Collectors;
+
+@Service
+public class CinemaSeatServiceImplementation implements CinemaSeatService {
+    @Autowired
+    CinemaSeatRepository cinemaSeatRepository;
+
+    @Autowired
+    ReservationService reservationService;
+
+    @Override
+    public CinemaSeat create(CinemaSeat cinemaSeat) {
+        return cinemaSeatRepository.save(cinemaSeat);
+    }
+
+    @Override
+    public CinemaSeat read(Long cinemaSeatId) {
+        Optional<CinemaSeat> cinemaSeatOptional = cinemaSeatRepository.findById(cinemaSeatId);
+        return cinemaSeatOptional.get();
+    }
+
+    @Override
+    public void update(Long cinemaSeatId, CinemaSeat cinemaSeat) {
+        cinemaSeatRepository.save(cinemaSeat);
+    }
+
+    @Override
+    public void delete(Long cinemaSeatId) {
+if(cinemaSeatRepository.existsById(cinemaSeatId)){
+    cinemaSeatRepository.deleteById(cinemaSeatId);
+}
+    }
+
+    @Override
+    public Iterable<CinemaSeat> getAllCinemaSeats() {
+        return cinemaSeatRepository.findAll();
+    }
+
+    @Override
+    public List<CinemaSeat> getCinemaSeatsByScreening(Screening screening) {
+        List<CinemaSeat> cinemaSeats = cinemaSeatRepository.findByScreening(screening);
+        cinemaSeats.sort(Comparator.comparing(seat -> seat.getSeatNumber()));
+        cinemaSeats.sort(Comparator.comparing(seat -> seat.getSeatRow()));
+        return cinemaSeats;
+    }
+
+    @Override
+    public List<CinemaSeat> findReservedSeats() {
+        return cinemaSeatRepository.findByReservationNotNull();
+    }
+
+    @Override
+    public boolean checkIfBlankBetweenReservedSeats(List<CinemaSeat> seatsList) {
```

Comment:
Ta funkcja zdecydowanie wymaga refaktoru :(

Reply:
Chyba za bardzo skupiłem się na tym żeby cała metoda działała zapominając o tym żeby może rozłożyć to na mniejsze części. Z rzeczy, które mógłbym zaproponować przy refaktoringu:
1. Pierwsza część metody, która tworzy listę temporarySeats i dodaje do niej odpowiednie miejsca mogła by być oddzielną metodą.
2. Pętla, która dla miejsc, które przyszły w rezerwacji ustawia rezerwację w temporarySeats mogła by być oddzielną metodą lub być częścią poprzedniej.
3.Sprawdzanie czy siedzenia z rezerwacji mogą być zarezerwowane, używając temporarySeats jako listy siedzeń w sali kinowej mogło by być oddzielną metodą.
4. Powyższe 3 metody używają temporarySeats. Rozważyłbym utworzenie takiej klasy.